server.js

```diff
@@ -734,13 +734,30 @@ app.post('/submit-match',  asyncMiddleware( async (req, res, next) => {
     var new_best_network_flag = false;
     var sgfbuffer = Buffer.from(req.files.sgf.data);
 
-    zlib.unzip(sgfbuffer, (err, sgfbuffer) => {
+    zlib.unzip(sgfbuffer, async (err, sgfbuffer) => {
```

Comment:
Maybe it doesn't matter but normally when needed we wrap this like

> zlib.unzip(networkbuffer,  asyncMiddleware( async (err, networkbuffer, next) => {

(Copied from elsewhere in the code.) And then appropriate closing backets later on. My limited understanding is that the ayncMidddleware is needed to pass error callbacks around properly?

Reply:
Yeah i assumed this would work based on what i know about await, but you are probably right about the error callback.

Reply:
`asyncMiddleware` looks like it's specifically for express where the callback needs to call the `next` middleware. `zlip.unzip`'s callback is only expecting a function that has 2 parameters: error, result.

Reply:
Ah ok. I think @rchs819 is re-writing this section of submit-match anyway. Maybe I should merge his first?

Reply:
Up to you, if you pull his i just rebase this one.